core/src/main/java/org/elasticsearch/rest/action/support/RestActions.java

```diff
@@ -93,6 +108,95 @@ public static void buildBroadcastShardsHeader(XContentBuilder builder, ToXConten
         builder.endObject();
     }
 
+    /**
+     * Create the XContent header for any {@link BaseNodesResponse}.
+     *
+     * @param builder XContent builder.
+     * @param params XContent parameters.
+     * @param response The response containing individual, node-level responses.
+     * @see #buildNodesHeader(XContentBuilder, Params, int, int, int, List)
+     */
+    public static <NodeResponse extends BaseNodeResponse> void buildNodesHeader(final XContentBuilder builder, final Params params,
+                                                                                final BaseNodesResponse<NodeResponse> response)
+        throws IOException {
+        final int successful = response.getNodes().size();
```

Comment:
Note that we also use this pattern for share responses which are in the same situation.